### ISSUE TITLE
Incorrect test for Promise.cast in polyfill()

### ISSUE THREAD

**Issue by jlatrive** [jakearchibald/es6-promise#128]
The polyfill function tests for Promise.cast to determine if it should automatically attach to the global environment.
I would rather use the environment's implementation if available, but as a result of this test, the native Promise object in Chrome, Firefox and node 0.12 gets replaced.
There is no mention of a cast method in the official spec at 
http://www.ecma-international.org/ecma-262/6.0/#sec-promise-objects so is there any reason for this test ?

**Comment by stefanpenner** [comment]
context: https://github.com/jakearchibald/es6-promise/issues/59

**Comment by jlatrive** [comment]
Thanks for the response, and sorry for the duplicate.

**Comment by stefanpenner** [comment]
> Thanks for the response, and sorry for the duplicate.

not a problem, I only know there is a duplicate because was part of the discussion.

TL;DR the resolution will be to just provide 2 different builds, one with and one without the polyfil mode.

It is tricky to satisfy everyones preference here, so we will just specialize and lets users choose .
